examples/src/createReactAgent/updateModelBeforeCall.ts

```diff
@@ -0,0 +1,557 @@
+/* eslint-disable no-nested-ternary */
+/* eslint-disable no-template-curly-in-string */
+/* eslint-disable no-plusplus */
+/**
+ * Update Model to Use Before Model Call
+ *
+ * This capability enables dynamic model selection based on context, allowing the agent to choose the most
+ * appropriate language model for each specific task or situation.
+ *
+ * Why this is important:
+ * - Cost Optimization: Uses smaller, cheaper models for simple tasks and reserves powerful models for complex reasoning
+ * - Performance Matching: Selects models with capabilities that match the specific requirements of each request
+ * - Specialized Routing: Directs different types of queries to models that excel in those particular domains
+ *
+ * Example Scenario:
+ * You're building a coding assistant that handles both simple syntax questions and complex algorithm design.
```

Comment:
question: we can access state, long term memory, and context before we update the model to use, right?